ephios/core/forms/events.py

```diff
@@ -210,20 +211,25 @@ def __init__(self, *args, **kwargs):
             ),
         )
 
+    def clean_meeting_time(self):
+        return from_current_timezone(
+            datetime.combine(self.cleaned_data["date"], self.cleaned_data["meeting_time"])
+        )
+
+    def clean_start_time(self):
+        return from_current_timezone(
+            datetime.combine(self.cleaned_data["date"], self.cleaned_data["start_time"])
+        )
+
+    def clean_end_time(self):
+        end_time = datetime.combine(self.cleaned_data["date"], self.cleaned_data["end_time"])
+        if make_aware(end_time) <= self.cleaned_data["start_time"]:
+            end_time += timedelta(days=1)
+        return from_current_timezone(end_time)
+
     def clean(self):
         cleaned_data = super().clean()
-        if {"date", "meeting_time", "start_time", "end_time"} <= set(cleaned_data.keys()):
-            cleaned_data["meeting_time"] = make_aware(
-                datetime.combine(cleaned_data["date"], cleaned_data["meeting_time"])
-            )
-            cleaned_data["start_time"] = make_aware(
-                datetime.combine(cleaned_data["date"], cleaned_data["start_time"])
-            )
-            cleaned_data["end_time"] = make_aware(
-                datetime.combine(self.cleaned_data["date"], cleaned_data["end_time"])
-            )
-            if self.cleaned_data["end_time"] <= self.cleaned_data["start_time"]:
-                cleaned_data["end_time"] = cleaned_data["end_time"] + timedelta(days=1)
+        if {"meeting_time", "start_time"} <= set(cleaned_data.keys()):
             if not cleaned_data["meeting_time"] <= cleaned_data["start_time"]:
```

Comment:
is it a good idea that we just shift the start time by one day but not the meeting time? (can also be a part of rethinking shifts with respect to >24h to get the bugfix out quickly)